arangod/Cluster/ClusterInfo.cpp

```diff
@@ -912,128 +957,148 @@ void ClusterInfo::loadCurrent() {
   // Now contact the agency:
   AgencyCommResult result = _agency.getValues(prefixCurrent);
 
-  if (result.successful()) {
-    velocypack::Slice slice = result.slice()[0].get(
-        std::vector<std::string>({AgencyCommManager::path(), "Current"}));
+  if (!result.successful()) {
+    LOG_TOPIC(DEBUG, Logger::CLUSTER)
+      << "Error while loading " << prefixCurrent
+      << " httpCode: " << result.httpCode() << " errorCode: " << result.errorCode()
+      << " errorMessage: " << result.errorMessage() << " body: " << result.body();
 
-    auto currentBuilder = std::make_shared<VPackBuilder>();
-    currentBuilder->add(slice);
+    return;
+  }
 
-    VPackSlice currentSlice = currentBuilder->slice();
+  auto slice = result.slice()[0].get( // get slice
```

Comment:
should we check if `result.slice()` is array?

Reply:
done